src/www/myInfo.js

```diff
@@ -0,0 +1,273 @@
+        subjectDiv.appendChild(paragraph);
+
+        for (const subject of subjectArray) {
+            const subjectPart = createListItem(subject.name + ": " + (subject.score * 10).toFixed(2));
+            subjectList.appendChild(subjectPart);
+            subjectDiv.appendChild(subjectList);
+        }
+    }
+
+    /**
+     * @summary Creates a button for each file in the data folder
+     * @param {string} innerText A string
+     * @returns {HTMLElement} Returns a html li element with the string sat as innerText
+     */
+    function createListItem(innerText) {
+        const listItem = document.createElement("li");
+        listItem.innerText = innerText;
+        return listItem;
+    }
+
+    /**
+     * @summary Creates a html list element with all the group members
+     * @param {object} group One group as a object
+     * @param {string} userName Name of the logged in user
+     */
+    function showGroup(group, userName) {
+        const groupMemberDiv = document.getElementById("currentGroup");
+        const button = document.getElementById("leaveButton");
+        const groupMembersList = document.createElement("ul");
+        const paragraph = document.createElement("h3");
+
+        clearChildren(groupMemberDiv);
+
+        for (const member of group.students) {
+            if (member.name !== userName) {
+                const memberName = createListItem(member.name);
+                groupMembersList.appendChild(memberName);
+            }
+        }
+        if (group.students.length === 1) {
+            const notInGroup = document.createElement("h3");
+            notInGroup.innerText = "Not in a group yet";
+            groupMemberDiv.appendChild(notInGroup);
+            button.style.display = "none";
+        }
+        if (group.students.length > 1) {
+            paragraph.innerText = "Your group " + group.name + ":";
+            button.style.display = "block";
+        }
+        groupMemberDiv.appendChild(paragraph);
+        groupMemberDiv.appendChild(groupMembersList);
+        if (group.students.length > 1) {
+            groupMemberDiv.appendChild(createGroupSubjectsElement(group));
+        }
+    }
+    /**
+     * @summary Creates a html list element with all the group members subjects added together
+     * @param {object} group One group as a object
+     * @returns {HTMLElement} Returns a H3 Element with a ul child element with the subjects from the group
+     */
+    function createGroupSubjectsElement(group) {
+        let counter = 0;
+        const paragraph = document.createElement("h3");
+        const groupSubjectList = document.createElement("ul");
+        const subjectArray = addSubjectScore(group);
+        /* Sorting the array, the bigest score will be in the first index of the array */
+        subjectArray.sort((a, b) => b.score - a.score);
+
+        paragraph.innerText = "Your group prefered 3 subjects:";
+
+        for (const subject of subjectArray) {
```

Comment:
Brug ikke en for of hvis du har brug for et index (counter i dette tilfælde) Brug et for in i stedet for

Reply:
I kan med fordel bruge slice her i stedet for sådan at i har et array der er 3 lang i stedet for det her gøjl

Reply:
fixed